finetrainers/trainer.py

```diff
@@ -751,13 +752,13 @@ def train(self) -> None:
                             )
                             accelerator.save_state(save_path)
 
-                # Maybe run validation
-                should_run_validation = (
-                    self.args.validation_every_n_steps is not None
-                    and global_step % self.args.validation_every_n_steps == 0
-                )
-                if should_run_validation:
-                    self.validate(global_step)
+                    # Maybe run validation
+                    should_run_validation = (
+                        self.args.validation_every_n_steps is not None
+                        and global_step % self.args.validation_every_n_steps == 0
+                    )
+                    if should_run_validation:
+                        self.validate(global_step)
```

Comment:
global_step is only incremented inside the `if accelerator.sync_gradients` statement. If we perform validation with `gradient_accumulation_steps > 1`, validation will run `gradient_accumulation_steps` many times until the next gradient sync happens. I believe this is correct approach